sensors/src/deleting_machines_sensor.py

```diff
@@ -0,0 +1,94 @@
+from datetime import datetime
+from st2reactor.sensor.base import PollingSensor
+from st2reactor.container.sensor_wrapper import SensorService
+from openstack_api.openstack_connection import OpenstackConnection
+
+
+class DeletingMachinesSensor(PollingSensor):
+    def __init__(self, sensor_service: SensorService, config, poll_interval):
+        # pylint: disable=super-with-arguments
+        super(DeletingMachinesSensor, self).__init__(
+            sensor_service=sensor_service, config=config, poll_interval=poll_interval
+        )
+        self.sensor_service: SensorService = sensor_service
+        self._logger = self.sensor_service.get_logger(name=self.__class__.__name__)
+
+    # pylint: disable=missing-function-docstring
+    def add_trigger(self, trigger):
+        pass
+
+    def update_trigger(self, trigger):
+        pass
+
+    def cleanup(self):
+        pass
+
+    def remove_trigger(self, trigger):
+        pass
+
+    def setup(self):
+        pass
+
+    def poll(self, cloud_account: str = "dev-admin"):
+        """
+        Action to check for machines that are stuck deleting for more than 10mins
+        Outputs a suitable dictionary to pass into create_tickets
+        """
+        output = {
+            "title": "Server {p[id]} has not been updated in more than 10 minutes during {p[action]}",
+            "body": "The following information may be useful\nHost id: {p[id]}",
+            "server_list": [],
+        }
+
+        with OpenstackConnection(cloud_name=cloud_account) as conn:
+            projects = conn.list_projects()
+        for project in projects:
+            deleted = self._check_deleted(project=project["id"], cloud=cloud_account)
+            print(output["server_list"])
+            output["server_list"].extend(deleted)
+        print(output["server_list"])
+        if len(output["server_list"]) > 0:
```

Comment:
A list with elements is "truthy" in Python so this is the equivalent and preferred in Python for future reference:
```python
if (output["server_list"]):
    self.sensor_service.dispatch_with_context(
````